core/src/main/java/org/elasticsearch/action/ingest/SimulateExecutionService.java

```diff
@@ -38,20 +39,38 @@
         this.threadPool = threadPool;
     }
 
+    void executeVerboseDocument(Processor processor, IngestDocument ingestDocument, List<SimulateProcessorResult> processorResultList) throws Exception {
```

Comment:
This method is kind of tricky now. I tried to come up with a different approach, but that didn't make it easier.